test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/HistoryTests.cs

```diff
@@ -194,52 +194,78 @@ public async Task GivenAValueForSinceAndBeforeWithModifications_WhenGettingSyste
-            Thread.Sleep(1000);
-            newResources.Add(await CreateResourceWithTag(Samples.GetJsonSample("BloodGlucose").ToPoco(), tag));
-            newResources.Add(await CreateResourceWithTag(Samples.GetJsonSample("BloodPressure").ToPoco(), tag));
-            newResources.Add(await CreateResourceWithTag(Samples.GetJsonSample("Patient-f001").ToPoco(), tag));
-            newResources.Add(await CreateResourceWithTag(Samples.GetJsonSample("Condition-For-Patient-f001").ToPoco(), tag));
+                    newResources.Add(await CreateResourceWithTag(Samples.GetDefaultPatient().ToPoco(), tag));
+                    newResources.Add(await CreateResourceWithTag(Samples.GetDefaultOrganization().ToPoco(), tag));
+                    await Task.Delay(1000);
+                    newResources.Add(await CreateResourceWithTag(Samples.GetJsonSample("BloodGlucose").ToPoco(), tag));
+                    newResources.Add(await CreateResourceWithTag(Samples.GetJsonSample("BloodPressure").ToPoco(), tag));
+                    newResources.Add(await CreateResourceWithTag(Samples.GetJsonSample("Patient-f001").ToPoco(), tag));
+                    newResources.Add(await CreateResourceWithTag(Samples.GetJsonSample("Condition-For-Patient-f001").ToPoco(), tag));
 
-            var sinceUriString = HttpUtility.UrlEncode(since.ToString("o"));
+                    var sinceUriString = HttpUtility.UrlEncode(since.ToString("o"));
 
-            // Query all the recent changes
-            using FhirResponse<Bundle> allChanges = await _client.SearchAsync($"_history?_tag={tag}&_since=" + sinceUriString);
+                    // Query all the recent changes
+                    using FhirResponse<Bundle> allChanges = await _client.SearchAsync($"_history?_tag={tag}&_since=" + sinceUriString);
 
-            AssertCount(7, allChanges.Resource.Entry);
+                    AssertCount(7, allChanges.Resource.Entry);
 
-            // now choose a value of before that is as close as possible to one of the last updated times
-            var lastUpdatedTimes = allChanges.Resource.Entry.Select(e => e.Resource.Meta.LastUpdated).OrderBy(d => d.Value);
+                    // now choose a value of before that is as close as possible to one of the last updated times
+                    var lastUpdatedTimes = allChanges.Resource.Entry.Select(e => e.Resource.Meta.LastUpdated).OrderBy(d => d.Value);
 
-            var before = lastUpdatedTimes.ToList()[4];
-            var beforeUriString = HttpUtility.UrlEncode(before.Value.ToString("o"));
-            Thread.Sleep(500);
-            var firstSet = await _client.SearchAsync($"_history?_tag={tag}&_since={sinceUriString}&_before={beforeUriString}");
+                    var before = lastUpdatedTimes.ToList()[4];
+                    var beforeUriString = HttpUtility.UrlEncode(before.Value.ToString("o"));
+                    await Task.Delay(500);
```

Comment:
>await Task.Delay(500); [](http://example.com/codeflow?start=20&length=22)

I don't understand why this delay is needed?

Reply:
Due to the possible time difference, without the delay, the server would complain that the time was in the future.